test/Microsoft.Extensions.Internal.Test/TypeNameHelperTest.cs

```diff
@@ -128,6 +142,66 @@ public void Can_pretty_print_array_name(string expected, Type type, bool fullNam
             Assert.Equal(expected, TypeNameHelper.GetTypeDisplayName(type, fullName));
         }
 
+        public static IEnumerable<object[]> GetOpenGenericsTestData()
+        {
+            yield return new object[] { "List<>", typeof(List<>), false };
+            yield return new object[] { "Dictionary<,>", typeof(Dictionary<,>), false };
+            yield return new object[] { "System.Collections.Generic.List<>", typeof(List<>) };
+            yield return new object[] { "System.Collections.Generic.Dictionary<,>",typeof(Dictionary<,>) };
+
+            yield return new object[]
+            {
+                "Microsoft.Extensions.Internal.TypeNameHelperTest+Level1<>+Level2<>+Level3<>",
+                typeof(Level1<>.Level2<>.Level3<>)
+            };
+
+            yield return new object[]
+            {
+                "Microsoft.Extensions.Internal.TypeNameHelperTest+OuterGeneric<>+InnerNonGeneric+InnerGeneric<,>+InnerGenericLeafNode<>",
+                typeof(OuterGeneric<>.InnerNonGeneric.InnerGeneric<,>.InnerGenericLeafNode<>)
+            };
+
+            var openDictionaryType = typeof(Dictionary<,>);
+            var genArgsDictionary = openDictionaryType.GetGenericArguments();
+            genArgsDictionary[0] = typeof(B<>);
+            var closedDictionaryType = openDictionaryType.MakeGenericType(genArgsDictionary);
+
+            yield return new object[]
+            {
+                "System.Collections.Generic.Dictionary<Microsoft.Extensions.Internal.TypeNameHelperTest+B<>,>",
+                closedDictionaryType
+            };
+
+            var openLevelType = typeof(Level1<>.Level2<>.Level3<>);
+            var genArgsLevel = openLevelType.GetGenericArguments();
+            genArgsLevel[1] = typeof(string);
+            var closedLevelType = openLevelType.MakeGenericType(genArgsLevel);
+
+            yield return new object[]
+            {
+                "Microsoft.Extensions.Internal.TypeNameHelperTest+Level1<>+Level2<string>+Level3<>",
```

Comment:
How does one construct such a type without reflection? I thought C# doesn't let you mix open and closed generic types this way.

Reply:
For case like this `class TestType<T> : BaseType<int,T> { ... }` result of `typeof(TestType).BaseType` will be type with one open and one closed parameter. C# doesn't allow mix generics only in `typeof` operator.

Reply:
I'll add one unit test for such type without reflection to make it more apparent.